examples/server.rs

```diff
@@ -0,0 +1,33 @@
+use futures::prelude::*;
+use serde_json::Value;
+use tokio::{
+    codec::{FramedRead, LengthDelimitedCodec},
+    net::TcpListener,
+};
+use tokio_serde::formats::Json;
+
+#[tokio::main]
+pub async fn main() {
+    // Bind a server socket
+    let listener = TcpListener::bind("127.0.0.1:17653").await.unwrap();
+
+    println!("listening on {:?}", listener.local_addr());
+
+    let mut s = listener.incoming();
+
+    while let Some(socket) = s.try_next().await.unwrap() {
+        // Delimit frames using a length header
+        let length_delimited = FramedRead::new(socket, LengthDelimitedCodec::new());
+
+        // Deserialize frames
+        let mut deserialized =
+            tokio_serde::FramedRead::<_, Value, _>::new(length_delimited, Json::default());
+
+        // Spawn a task that prints all received messages to STDOUT
+        tokio::spawn(async move {
+            while let Some(msg) = deserialized.try_next().await.unwrap() {
```

Comment:
I have tried this with bincode and messagepack as well and the TryStream trait isn't happy anymore as in:
```
    |
28  |         match deserialized.try_next().await {
    |                            ^^^^^^^^ method not found in `tokio_serde::FramedRead<tokio_codec::framed_read::FramedRead<tokio_net::tcp::stream::TcpStream, tokio_codec::length_delimited::LengthDelimitedCodec>, PacketHeader, tokio_serde::formats::bincode::Bincode<_>>`
    |
    = note: the method `try_next` exists but the following trait bounds were not satisfied:
            `tokio_serde::FramedRead<tokio_codec::framed_read::FramedRead<tokio_net::tcp::stream::TcpStream, tokio_codec::length_delimited::LengthDelimitedCodec>, PacketHeader, tokio_serde::formats::bincode::Bincode<_>> : futures_util::try_stream::TryStreamExt`

```

Reply:
It was caused by Error type mismatch. Should work now.